controller.go

```diff
@@ -469,3 +457,71 @@ func upgradePVFrom12(pv *PV) (deleted bool, err error) {
 	}
 	return false, nil
 }
+
+func provisionClaim(pvc *api.PersistentVolumeClaim) error {
+	// Provisioning can take some time. Start it in a separate goroutine.
+	// Keep a map of running provisioners and start new provisioning only if
+	// it's not already running. There will be some locks around, these are
+	// left out to implementation
+
+	if !runningProvisioners.isRunning(pvc) {
```

Comment:
agree.  i think a provisioner would 'check out' a claim by populating the provisionerStarted field.  Other provisionrs would have to respect that for provisionerStarted + 15 min or something